transport/src/main/java/io/netty/channel/pool/FixedChannelPool.java

```diff
@@ -441,10 +442,19 @@ public void acquired() {
 
     @Override
     public void close() {
+        closeAsync().awaitUninterruptibly();
+    }
+
+    /**
+     * Closes the pool in an async manner.
+     *
+     * @return Future which represents completion of the close task
+     */
+    public Future<?> closeAsync() {
         if (executor.inEventLoop()) {
-            close0();
+            return close0();
         } else {
-            executor.submit(new Runnable() {
+            return executor.submit(new Runnable() {
                 @Override
                 public void run() {
                     close0();
```

Comment:
hmm... close0 returns a future....  we need to do something with it

Reply:
Thanks for catching. Fixed it.